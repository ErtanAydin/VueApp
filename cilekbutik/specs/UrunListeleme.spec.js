import { mount, createLocalVue, RouterLinkStub } from "@vue/test-utils";
import Vuex from "vuex"
import UrunListeleme from "@/components/UrunListeleme";

const localVue = createLocalVue()
localVue.use(Vuex)

const store = new Vuex.Store({
  state: {
    Urunler: [],
    deger:0
  },
  actions: {
    Urunlerfetch(){
      let Urunler = [
        {
            id:0,
            url:'https://www.cilekbutik.net/cdn/1/300/500/images/urunler/5fa57c217b440-1353.jpg',
            name:'Lakers Sweat',
            fiyat:55.00,
            url2:'https://www.cilekbutik.net/cdn/2/100/100/images/urunler/5fa6a282db005-1368.jpg',
            barkod:"6242005630101",
            kategori:"Ceket",
            bedenler:[{ad:'S'},{ad:'M'},{ad:'L'}],
            beden:'M',
            renk:'TURUNCU',
            tutar:55.00,
            adet:10,
        }]
        this.state.Urunler=Urunler
    },
    SepeteEkle(){
      let deger = 0
      this.state.deger = deger
    }
  }
})

describe("UrunListeleme", () => {
  test("mounts properly", () => {
    const wrapper = UrunListeleme;
    expect(wrapper).toBeTruthy();
  });

  test("gerÃ§ek Vuex Store kullanarak UrunListeleme render edildi", () => {
    const wrapper = mount(UrunListeleme, { 
      store, 
      localVue,
      stubs: {
        NuxtLink: RouterLinkStub
      } 
    })

    expect(wrapper.html()).toMatchSnapshot()
  });
});

describe('created', () => {
  test('created test', () => {
    expect(store.state.Urunler).not.toBe(undefined)
  })
})

describe('computed', () => {
  test('computed test', () => {
    expect(store.state.Urunler).not.toBe(undefined)
  })
})

describe('methods', () => {
  test('methods test', () => {
    const wrapper = mount(UrunListeleme, { 
      store, 
      localVue,
      stubs: {
        NuxtLink: RouterLinkStub
      } 
    })
    wrapper.vm.SepeteEkle()
    expect(store.state.deger).toBe(0)
  })
})