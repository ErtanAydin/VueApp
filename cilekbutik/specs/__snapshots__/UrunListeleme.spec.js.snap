// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UrunListeleme gerçek Vuex Store kullanarak UrunListeleme render edildi 1`] = `
<div>
  <div class="UrunAlanı"><a><img title="Lakers Sweat" src="https://www.cilekbutik.net/cdn/1/300/500/images/urunler/5fa57c217b440-1353.jpg" alt="" class="UrunImage"></a> <a class="LINK">
      <p title="Lakers Sweat" class="UrunIsim"> Lakers Sweat</p>
    </a> <a class="LINK"><b title="Lakers Sweat" class="UrunFiyat"><span>55,00 TL </span></b></a> <button class="SepeteEkleButton"> Sepete Ekle </button></div>
</div>
`;
